File: TesteAccLogin2/app/src/main/java/com/example/testeacclogin/ui/login/LoginViewModelFactory.java
package com.example.testeacclogin.ui.login;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.example.testeacclogin.data.LoginDataSource;
import com.example.testeacclogin.data.LoginRepository;

/**
 * ViewModel provider factory to instantiate LoginViewModel.
 * Required given LoginViewModel has a non-empty constructor
 */
public class LoginViewModelFactory implements ViewModelProvider.Factory {

    @NonNull
    @Override
    @SuppressWarnings("unchecked")
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(LoginViewModel.class)) {
            return (T) new LoginViewModel(LoginRepository.getInstance(new LoginDataSource()));
        } else {
            throw new IllegalArgumentException("Unknown ViewModel class");
        }
    }
}

File: BankAndroid/app/src/main/java/br/com/bank/android/login/factory/LoginViewModelFactory.kt
package br.com.bank.android.login.factory

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import br.com.bank.android.login.business.LoginBusiness
import br.com.bank.android.login.handler.LoginHandler
import br.com.bank.android.login.presentation.model.LoginViewModel

class LoginViewModelFactory(
    private val loginBusiness: LoginBusiness,
    private val loginHandler: LoginHandler
) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return LoginViewModel(loginBusiness, loginHandler) as T
    }
}